### PR TITLE
Add .cirrus.yml config for Cirrus-CI continuous integration

### PR DESCRIPTION
**Pull request by emaste**
Starting with Alpine and FreeBSD 11 and 12 tasks, others to be added.

Result from my fork: https://cirrus-ci.com/build/5686867601653760

### CODE REVIEW

**Comment by z3ntu** [comment]
See also #6

**Comment by Youw** [comment]
Even with #6, this does looks like an improvement. @z3ntu or do you suggest to merge the effort, or smth?

Also, in order to actually have this working, someone from libusb organisation has to give access to Cirrus-CI to this repository.

**Comment by z3ntu** [comment]
I'd rather use the builds.sr.ht service as it's open-source and could  theoretically be self-hosted (and I doubt cirrus ci is open-source).

**Comment by Qbicz** [comment]
Since we already have Alpine and FreeBSD, I would suggest to close this PR.

@z3ntu do you see value in having Cirrus configuration available? Or in testing multiple versions of FreeBSD as @emaste did?

**Comment by z3ntu** [comment]
builds.sr.ht also has support for FreeBSD 11.x and FreeBSD CURRENT if we want that. Our freebsd builds use `freebsd/latest` which is FreeBSD 12 currently. So if you want that, I can add them.

But I think we're good with the current set of CIs (Alpine Linux with musl libc; Arch Linux with glibc, FreeBSD and three variants of Windows (mingw, cygwin & msbuild))

What could be done is to create "release candidate" CI configuration that makes a build for each OS supported on builds.sr.ht (Alpine Linux, Arch Linux, Debian, Fedora, FreeBSD, NixOS, OpenBSD, Ubuntu) and all versions supported there if someone has too much free time on their hands^^

**Comment by emaste** [comment]
builds.sr.ht sounds like a good approach, although IMO there's not much downside of having a configuration for Cirrus-CI in the tree also, as with .travis.yml / .appveyor.yml; people doing development on their fork may find it convenient.

**Comment by Qbicz** [comment]
@youw, @z3ntu, I support merging this if you agree. We don't have to have Cirrus set up for this repo, this could be for contributors convenience.

File: .cirrus.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+alpine_task:
+        container:
+                image: alpine:latest
+        install_script: apk add autoconf automake g++ gcc libusb-dev libtool linux-headers eudev-dev make musl-dev
+        script:
+                - ./bootstrap
+                - ./configure || { cat config.log; exit 1; }
```
Comment (z3ntu) [inline .cirrus.yml]: Why does this configure statement have the statement behind but not the others?

Comment (emaste) [reply .cirrus.yml]: Oh, I probably added that while trying to track down some problem building on alpine, and did not run into a problem on the other OSes that required more details. It's probably a good idea for all cases and I'll update the PR.

**Comment by emaste** [comment]
Here's an Alpine, FreeBSD 11, FreeBSD 12 build: https://cirrus-ci.com/build/4977906178064384

**Comment by Youw** [comment]
One last note before merge: can we move this file to a `.builds` subfolder?

**Comment by emaste** [comment]
> One last note before merge: can we move this file to a `.builds` subfolder?

No, Cirrus-CI only looks at the root dir of the repository (as does Travis-CI etc.)

**Comment by Youw** [comment]
I see, OK, let it be then.

**Pull request merged**